console-subscriber/src/aggregator.rs

```diff
@@ -67,15 +82,56 @@ pub(crate) struct Flush {
     pub(crate) triggered: AtomicBool,
 }
 
-struct Stats {
-    // task stats
-    polls: u64,
+// An entity that at some point in time can be closed
+// and recycled
+trait Closable {
+    fn closed_at(&self) -> &Option<SystemTime>;
+}
+
+trait ToProto {
+    type Result;
+    fn to_proto(&self) -> Self::Result;
```

Comment:
nit: can we call the associated type something other than `Result`? `Result` makes me think of the `Result` type, but this conversion is infallible.

how about either
```suggestion
    type Proto;
    fn to_proto(&self) -> Self::Proto;
```
or
```suggestion
    type Output;
    fn to_proto(&self) -> Self::Output;
```